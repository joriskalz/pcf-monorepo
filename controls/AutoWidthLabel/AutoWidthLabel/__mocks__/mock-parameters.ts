/* istanbul ignore file */

import { IInputs } from "../generated/ManifestTypes";
import {
  MockDecimalNumberProperty,
  MockEnumProperty,
  MockStringProperty,
  MockWholeNumberProperty,
} from "./mock-context";

export function getMockParameters(): IInputs {
  return {
    Text: new MockStringProperty(),
    PaddingLeft: new MockWholeNumberProperty(),
    PaddingRight: new MockWholeNumberProperty(),
    PaddingTop: new MockWholeNumberProperty(),
    PaddingBottom: new MockWholeNumberProperty(),
    FontName: new MockStringProperty(),
    FontSize: new MockDecimalNumberProperty(),
    FontSizeUnits: new MockEnumProperty("1"),
    FontColor: new MockStringProperty(),
    FontWeight: new MockStringProperty(),
    DisabledFontColor: new MockStringProperty(),
    DisabledFontWeight: new MockStringProperty(),
    FocusFontColor: new MockStringProperty(),
    FocusFontWeight: new MockStringProperty(),
    HoverFontColor: new MockStringProperty(),
    HoverFontWeight: new MockStringProperty(),
    FillColor: new MockStringProperty(),
    DisabledFillColor: new MockStringProperty(),
    FocusFillColor: new MockStringProperty(),
    HoverFillColor: new MockStringProperty(),
    //DisabledHoverFillColor: new MockStringProperty(),
    BorderColor: new MockStringProperty(),
    BorderThickness: new MockWholeNumberProperty(),
    BorderRadius: new MockWholeNumberProperty(),
    DisabledBorderColor: new MockStringProperty(),
    FocusBorderColor: new MockStringProperty(),
    FocusBorderThickness: new MockWholeNumberProperty(),
    HoverBorderColor: new MockStringProperty(),
    HoverBorderThickness: new MockWholeNumberProperty(),
    //RenderBorderStyle: new MockEnumProperty<"0" | "1">(),
  };
}
