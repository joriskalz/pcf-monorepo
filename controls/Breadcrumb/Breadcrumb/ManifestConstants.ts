export const enum ManifestPropertyNames {
    dataset = 'dataset',
}

export const enum ItemColumns {
    DisplayName = 'ItemDisplayName',
    Key = 'ItemKey',
    Clickable = 'ItemClickable',
}

export const enum InputEvents {
    SetFocus = 'SetFocus',
}

export const enum InputProperties {
    InputEvent = 'InputEvent',
    SelectedKey = 'SelectedKey',
}
