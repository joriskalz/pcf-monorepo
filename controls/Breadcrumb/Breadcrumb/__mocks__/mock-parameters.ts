/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        items: new MockDataSet([]),
        MaxDisplayedItems: new MockWholeNumberProperty(),
        OverflowIndex: new MockWholeNumberProperty(),
    };
}
