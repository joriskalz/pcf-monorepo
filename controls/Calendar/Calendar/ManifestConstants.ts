export const enum InputEvents {
    SetFocus = 'SetFocus',
}
