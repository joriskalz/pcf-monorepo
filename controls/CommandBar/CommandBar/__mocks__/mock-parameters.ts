/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        Theme: new MockStringProperty(),
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        items: new MockDataSet([]),
    };
}
