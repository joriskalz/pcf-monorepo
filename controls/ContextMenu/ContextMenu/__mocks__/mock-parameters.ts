/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        TextAlignment: new MockEnumProperty(),
        items: new MockDataSet([]),
        Chevron: new MockTwoOptionsProperty(),
        FillColor: new MockStringProperty(),
        BorderColor: new MockStringProperty(),
        BorderRadius: new MockWholeNumberProperty(),
        FontColor: new MockStringProperty(),
        FontSize: new MockWholeNumberProperty(),
        HoverBorderColor: new MockStringProperty(),
        HoverFillColor: new MockStringProperty(),
        HoverFontColor: new MockStringProperty(),
        HoverIconColor: new MockStringProperty(),
        IconColor: new MockStringProperty(),
        IconSize: new MockWholeNumberProperty(),
    };
}
