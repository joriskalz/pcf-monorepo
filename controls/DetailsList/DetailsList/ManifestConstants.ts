export const enum RecordsColumns {
    RecordKey = 'RecordKey',
    RecordCanSelect = 'RecordCanSelect',
    RecordSelected = 'RecordSelected',
}

export const enum ColumnsColumns {
    ColDisplayName = 'ColDisplayName',
    ColName = 'ColName',
    ColWidth = 'ColWidth',
    ColSortable = 'ColSortable',
    ColSortBy = 'ColSortBy',
    ColHorizontalAlign = 'ColHorizontalAlign',
    ColVerticalAlign = 'ColVerticalAlign',
    ColMultiLine = 'ColMultiLine',
    ColIsBold = 'ColIsBold',
    ColTagColorColumn = 'ColTagColorColumn',
    ColTagBorderColorColumn = 'ColTagBorderColorColumn',
    ColResizable = 'ColResizable',
    ColHeaderPaddingLeft = 'ColHeaderPaddingLeft',
    ColCellType = 'ColCellType',
    ColShowAsSubTextOf = 'ColShowAsSubTextOf',
    ColPaddingTop = 'ColPaddingTop',
    ColPaddingLeft = 'ColPaddingLeft',
    ColLabelAbove = 'ColLabelAbove',
    ColMultiValueDelimiter = 'ColMultiValueDelimiter',
    ColFirstMultiValueBold = 'ColFirstMultiValueBold',
    ColInlineLabel = 'ColInlineLabel',
    ColHideWhenBlank = 'ColHideWhenBlank',
    ColSubTextRow = 'ColSubTextRow',
    ColAriaTextColumn = 'ColAriaTextColumn',
    ColCellActionDisabledColumn = 'ColCellActionDisabledColumn',
    ColImageWidth = 'ColImageWidth',
    ColImagePadding = 'ColImagePadding',
    ColRowHeader = 'ColRowHeader',
}

export const enum InputProperties {
    InputEvent = 'InputEvent',
}

export const enum InputEvents {
    SetFocus = 'SetFocus',
    SetFocusOnRow = 'SetFocusOnRow',
    SetFocusOnRowSetSelection = 'SetFocusOnRowSetSelection',
    ClearSelection = 'ClearSelection',
    SetSelection = 'SetSelection',
    SetFocusOnHeader = 'SetFocusOnHeader',
    LoadNextPage = 'LoadNextPage',
    LoadPreviousPage = 'LoadPreviousPage',
    LoadFirstPage = 'LoadFirstPage',
}

export const enum OutputEvents {
    Sort = 'Sort',
    CellAction = 'CellAction',
    OnRowSelectionChange = 'OnRowSelectionChange',
}

export const enum SortDirection {
    Ascending = '0',
    Descending = '1',
}

export const enum CellTypes {
    Expand = 'expand',
    Tag = 'tag',
    Image = 'image',
    ClickableImage = 'clickableimage',
    IndicatorTag = 'indicatortag',
    Link = 'link',
}
