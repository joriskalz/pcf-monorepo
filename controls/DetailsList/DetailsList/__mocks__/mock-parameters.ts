/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        Compact: new MockTwoOptionsProperty(),
        CurrentSortColumn: new MockStringProperty(),
        CurrentSortDirection: new MockEnumProperty('0'),
        PageSize: new MockWholeNumberProperty(),
        RaiseOnRowSelectionChangeEvent: new MockTwoOptionsProperty(),
        SelectRowsOnFocus: new MockTwoOptionsProperty(),
        SelectionType: new MockEnumProperty('0'),
        AlternateRowColor: new MockStringProperty(),
        HeaderVisible: new MockTwoOptionsProperty(false),
        SelectionAlwaysVisible: new MockTwoOptionsProperty(false),
        records: new MockDataSet([]),
        columns: new MockDataSet([]),
        LargeDatasetPaging: new MockTwoOptionsProperty(),
    };
}
