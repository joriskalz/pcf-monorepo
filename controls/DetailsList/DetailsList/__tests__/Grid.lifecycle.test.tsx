import { IInputs } from '../generated/ManifestTypes';

import { MockContext, MockState } from '../__mocks__/mock-context';
import { MockColumn, MockDataSet, MockEntityRecord } from '../__mocks__/mock-datasets';
import { getMockParameters } from '../__mocks__/mock-parameters';
import { FluentDetailsList } from '..';
import { ColumnsColumns, RecordsColumns } from '../ManifestConstants';

// Since requestAnimationFrame does not exist in the test DOM, mock it
window.requestAnimationFrame = jest.fn().mockImplementation((callback) => {
    callback();
});
jest.useFakeTimers();

describe('FluentDetailsList', () => {
    beforeEach(() => jest.clearAllMocks());
    it('renders', () => {
        const { component, context, notifyOutputChanged } = createComponent();
        component.init(context, notifyOutputChanged);
        const element = component.updateView(context);
        expect(element).toMatchSnapshot();
    });
});

function createComponent() {
    const component = new FluentDetailsList();
    const notifyOutputChanged = jest.fn();
    const context = new MockContext<IInputs>(getMockParameters());

    const records = [
        new MockEntityRecord('1', {
            [RecordsColumns.RecordKey]: '1',
            ['name']: 'Row 1',
        }),
    ];

    context.parameters.records = new MockDataSet(records);
    context.parameters.records.columns = [new MockColumn('name', 'Name')];
    context.parameters.records.getSelectedRecordIds = jest.fn().mockReturnValue([]);
    const columns = [
        new MockEntityRecord('1', {
            [ColumnsColumns.ColName]: 'name',
            [ColumnsColumns.ColDisplayName]: 'Name',
            [ColumnsColumns.ColWidth]: '100',
        }),
    ];

    context.parameters.columns = new MockDataSet(columns);

    const state = new MockState();
    const container = document.createElement('div');
    document.body.appendChild(container);
    return { component, context, notifyOutputChanged, container, state };
}
