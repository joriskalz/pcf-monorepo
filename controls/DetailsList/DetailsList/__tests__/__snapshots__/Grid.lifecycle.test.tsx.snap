// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FluentDetailsList renders 1`] = `
<Memo(Grid)
  columns={
    Object {
      "1": MockEntityRecord {
        "getNamedReference": [MockFunction],
        "id": "1",
        "values": Object {
          "ColDisplayName": "Name",
          "ColName": "name",
          "ColWidth": "100",
        },
      },
    }
  }
  compact={false}
  componentRef={[Function]}
  datasetColumns={
    Array [
      MockColumn {
        "displayName": "Name",
        "name": "name",
      },
    ]
  }
  height={-1}
  isHeaderVisible={true}
  onCellAction={[Function]}
  onNavigate={[Function]}
  onSort={[Function]}
  overlayOnSort={false}
  pageSize={null}
  records={
    Object {
      "1": MockEntityRecord {
        "getNamedReference": [MockFunction],
        "id": "1",
        "values": Object {
          "RecordKey": "1",
          "name": "Row 1",
        },
      },
    }
  }
  selectOnFocus={false}
  selection={
    Selection {
      "_anchoredIndex": 0,
      "_canSelectItem": [Function],
      "_changeEventSuppressionCount": 0,
      "_exemptedCount": 0,
      "_exemptedIndices": Object {},
      "_getKey": [Function],
      "_isModal": false,
      "_items": Array [],
      "_keyToIndexMap": Object {},
      "_onItemsChanged": undefined,
      "_onSelectionChanged": [Function],
      "_selectedItems": null,
      "_unselectableCount": 0,
      "_unselectableIndices": Object {},
      "count": 0,
      "mode": 2,
    }
  }
  selectionAlwaysVisible={false}
  selectionType={0}
  shimmer={false}
  sortedColumnIds={
    Array [
      "1",
    ]
  }
  sortedRecordIds={
    Array [
      "1",
    ]
  }
  sorting={
    Array [
      Object {
        "name": "",
        "sortDirection": "0",
      },
    ]
  }
  visible={true}
  width={-1}
/>
`;
